test/ReporterTest.cpp

```diff
@@ -239,17 +270,256 @@ ReporterTest::ReporterTest()
                                                  "",
                                                  true);
     m_reporter->init();
+
+    // ApplicationIO calls: to be removed
+    EXPECT_CALL(m_application_io, report_name()).WillOnce(Return(m_report_name));
+    EXPECT_CALL(m_application_io, profile_name());
```

Comment:
Please provide .Will*(Return(...)) for expectations on functions that return parameters.  This will reduce the warning output from running the test.